### DISCUSSION TITLE
UseSWR mutation

### DISCUSSION THREAD

**Discussion by HakkaOfDev** [vercel/community#677] [Help]
Hello,
I'm making changes on the blockchain and I'm rendering something with the useSWR hook. But I need someway and somehow to rendering my state even if there's changes. What I mean is, I'm pushing the changes on the blockchain but I need to wait 2-3 seconds from the API where I'm getting the changes, so temporary I'm using a timeout function with mutate() inside it. But maybe there is a way to fetch the data until there is a modification, I don't really know ^^

Thanks

**Comment by Babbili** [comment]
@HakkaOfDev use [ISR data fetching](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) ( Incremental Static Regeneration ), Next Js will update your page by re-validating the data fetched every 5 seconds window if you provided the `revalidate` prop as `5` that you add to `getStaticProps()` as below
```
function Page({ data }) {
  return (
    <>
         <Component />
    </>
  )
}
export default Page

// This function gets called at build time on server-side.
// It may be called again, on a serverless function, if
// revalidation is enabled and a new request comes in
export async function getStaticProps() {
  const res = await fetch('https://.../api')
  const data = await res.json()

  return {
    props: {
      data,
    },
    // Next.js will attempt to re-generate the page:
    // - When a request comes in
    // - At most once every 10 seconds
    revalidate: 10, // In seconds
  }
}
```
#677
